### PR TITLE
Feature/normalize css

### PR DESCRIPTION
**Pull request by andrewvallejo**
# Normalize CSS

- [x] New feature 
- [x] Refactor
- [x] Bugfix 
- [ ] Documentation

## About
This is a whoppin' PR! At first, it was an attempt to normalize CSS for mobile views across every page. This proved to be very difficult since we have an active footer and header (always visible). Every view has been tested and works well for mobile (I can't say the same for desktop).  A lot of restructuring the form flow. It wasn't that the code was bad, but it was difficult to jump into, especially with hopes of establishing a global context. The only thing I found that I might have eff'd up would be that I can't get the original fetch call to establish the charID to work. It keeps going undefined despite it working before. I think my new logic must've skipped something that @LKessell wrote. 

## Details
- Refactor all form pages
- Update JSX for most sections
- Update CSS for nearly everything to work well on mobile (tested on iPhone 12)
- Fix NavBar elements and refactored the Dice button
- Add Dice Page and dice/amount section
- dice/amount section CSS done
- Characteristics bubbles now use grid logic 
- Add @LKessell CSS to the rest of the Forms
- Navbar hides when in creation view. 
- NavBar only appears when a creator is created and when not in creation view

## Concerns

- Failed CircleCI
- I might have ripped too many cables from the brain of the robot... I would suggest not merging this until we can talk about it some more. 
- Unable to set charID in the first form (everything else works if you hardcode in 1)

### CODE REVIEW

File: src/components/FormCharDetails.js [hunk 1]
```diff
@@ -32,82 +33,107 @@ const CREATE_DETAILS = gql`
     ) {
       id
       name
+      species
```
Comment (LKessell) [inline src/components/FormCharDetails.js:36]: Adding the career, specialization, etc. here should allow you to access their values in the mutation response!

File: src/components/FormCharDetails.js [hunk 2]
```diff
@@ -32,82 +33,107 @@ const CREATE_DETAILS = gql`
...
   const [state, dispatch] = useReducer(formReducer, initialState);
+  const { state: userState, dispatch: userDispatch } = useContext(UserContext)
+  const { name, species, specialization, career, age, height, build, hair, eyes } = state;
+  
+  const [createCharDetails, { loading, error, data }] = useMutation(CREATE_DETAILS);
+
+  //  useEffect(() => {
+  //   createCharDetails({ 
+  //     variables: state 
+  //   }) // So this useEffect works every time the state is change (ever keypress) but it creates a new character per key press
```
Comment (LKessell) [inline src/components/FormCharDetails.js:52]: UUGGGHHHH 😂

**Comment by LKessell** [comment]
I think Circle might be failing because that test isn't intercepted/stubbed yet, so it's trying to make a character query without an ID being set first 🤔

File: src/components/FormCharDetails.js (hunk 1 above)
Comment (andrewvallejo) [reply src/components/FormCharDetails.js]: Gotchu, thanks!

**Pull request merged**